src/rgw/rgw_sal_motr.cc

```diff
@@ -4136,6 +4432,46 @@ int MotrStore::init_metadata_cache(const DoutPrefixProvider *dpp,
   return 0;
 }
 
+void MotrStore::addb_map_init(void)
+{
+  for (uint64_t idx = 0; idx < g_rgw_to_addb_idx_func_name_map_size; idx++) {
+    addb_func_to_idx_map[g_rgw_to_addb_idx_func_name_map[idx]] = idx;
+#if defined ADDB_DEBUG
+    ldout(cctx, 0) << "[MM] Registering func: <" << g_rgw_to_addb_idx_func_name_map[idx] << "> with ID: " << idx << dendl;
+#endif
+  }
+}
+
+uint64_t MotrStore::addb_func_to_idx(const std::string& funcname)
+{
+  size_t end = funcname.find('(');
+  size_t begin = funcname.find_last_of(' ', end) + 1;
+  string func = funcname.substr(begin, end - begin);
+  uint64_t idx;
+
+#if defined ADDB_DEBUG
+  ldout(cctx, 0) << "ADDB Debug: Func: " << func << dendl;
+  ldout(cctx, 0) << "ADDB Debug: Funcname: " << funcname << dendl;
+
+  for (const auto& v : addb_func_to_idx_map) {
+    ldout(cctx, 0) << "ADDB Debug: func:<"  << v.first << ">, idx:<" << v.second << ">" << dendl;
+  }
+#endif
+
+  if (addb_func_to_idx_map.count(func) == 0) {
+    ldout(cctx, 0) << "unknown ADDB function: " << func << dendl;
+    idx = 0;
+  } else {
+    idx = addb_func_to_idx_map[func];
```

Comment:
It was an opposite intention - to avoid static list (e.g., enum, array) of function and give more flexibility. By using PRETTY_FUNCTION it's impelemented partially - we still maintain array for idx->function_name_str conversion.
Having enum requires developer to pass exact corresponding value for a function, and it's a place of error.
Do you have alternative view on it? I assumed PRETTY_FUNCTION macro and map lookup doesn't affect performance, but what are your ideas? 
Let's schedule a call and discuss.